### PR TITLE
fix aarch64 ncat flake

### LINKED ISSUE #433: aarch64 flake: iptables - port forwarding ipv4 - sctp
```
not ok 48 firewalld - port forwarding ipv4 - sctp
# (from function `assert' in file test/helpers.bash, line 282,
#  from function `run_nc_test' in file test/helpers.bash, line 561,
#  from function `test_port_fw' in file test/helpers.bash, line 489,
#  in test file test/200-bridge-firewalld.bats, line 231)
#   `test_port_fw proto=sctp' failed
#  nsenter -n -m -w -t 9513 ip link set lo up
#  nsenter -n -m -w -t 9513 dbus-daemon --address=unix:path=/tmp/netavark_bats.xyCUHe/netavark-firewalld --print-pid --config-file=/var/tmp/netavark/test/testfiles/firewalld-dbus.conf
# 9524
# firewalld pid: 9525
#  nsenter -n -m -w -t 9513 firewall-cmd --state
# not running
# [ rc=252 ]
#  nsenter -n -m -w -t 9513 firewall-cmd --state
# running
# {
#   "container_id": "jRjOH1qLZBfsQizL9OBdJjIDlqGakNjhHoVDsXD4dlvIfn54G0GU8anz6uBAFRzO",
#   "container_name": "name-OQLhfplYp6",
#   "port_mappings": [
#     {
#       "host_ip": "",
#       "container_port": 14140,
#       "host_port": 13895,
#       "range": 1,
#       "protocol": "sctp"
#     }
#   ],
#   "networks": {
#     "podman1": {
#       "static_ips": [
#         "10.107.37.189"
#       ],
#       "interface_name": "eth0"
#     }
#   },
#   "network_info": {
#     "podman1": {
#       "name": "podman1",
#       "id": "ed82e3a703682a9c09629d3cf45c1f1e7da5b32aeff3faf82837ef4d005356e6",
#       "driver": "bridge",
#       "network_interface": "podman1",
#       "subnets": [
#         {"subnet":"10.107.37.0/24","gateway":"10.107.37.1"}
#       ],
#       "ipv6_enabled": true,
#       "internal": false,
#       "dns_enabled": true,
#       "ipam_options": {
#         "driver": "host-local"
#       }
#     }
#   }
# }
#  nsenter -n -m -w -t 9513 ./bin/netavark setup /proc/9515/ns/net
# {"podman1":{"dns_search_domains":["dns.podman"],"dns_server_ips":["10.107.37.1"],"interfaces":{"eth0":{"mac_address":"b2:d9:aa:05:97:68","subnets":[{"gateway":"10.107.37.1","ipnet":"10.107.37.189/24"}]}}}}
#  nsenter -n -m -w -t 9513 nc -4 --sctp 10.107.37.1 13895
# #/vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
# #|     FAIL: ncat received data
# #| expected: 'riNSXmdQSy'
# #|   actual: ''
# #\^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

I have seen this twice now, just filling this as issue so we can keep track.

https://cirrus-ci.com/task/5590240982204416
https://cirrus-ci.com/task/6384006940852224

### PR DESCRIPTION
**Pull request by Luap99**
I am not not 100% sure because I was unable to reproduce even with
hack/get_ci_vm.sh. This flakes permanently in CI so there must be a
different in the environment which causes it.

However looking at the code we spawn the nc listener in the background
so there is no guarantee that it will already listen when we make the nc
connect call. To fix this we use the wait_for_port logic to ensure the
port is bound.
For now there is no sctp support, the /proc file format for it is
completely different and I didn't want to spend more time on it so I
just added a sleep and call it good enough.

Fixes #433

### CODE REVIEW

**Comment by Luap99** [comment]
@mheon @baude @flouthoc PTAL
I didn't flake on this PR which is a success given that I had to restart it 4-6 times on other PRs until it passed.

**Review by flouthoc — APPROVED** [review]
LGTM

File: test/helpers.bash [hunk 1]
```diff
@@ -538,22 +525,47 @@ function is_ipv4() {
...
     nsenter -n -t "${CONTAINER_NS_PIDS[$container_ns]}" timeout --foreground -v --kill=10 5 \
         nc $nc_common_args -l -p $container_port &>"$NETAVARK_TMPDIR/nc-out" <$stdin &
 
+    # make sure to wait until port is bound otherwise test can flake
+    # https://github.com/containers/netavark/issues/433
+    if [ "$proto" = "tcp" ] || [ "$proto" = "udp" ]; then
+        wait_for_port "${CONTAINER_NS_PIDS[$container_ns]}" $container_port $proto
+    else
+        #TODO add support for sctp port reading from /proc/net/sctp/eps
+        # for new just sleep
```
[the remark this answers is not in the document]
Comment (Luap99) [reply test/helpers.bash]: thanks, fixed

**Pull request merged**